canonicalwebteam/blog/wordpress.py

```diff
@@ -82,22 +85,29 @@ def get_articles(
 
         :returns: response, metadata dictionary
         """
-        response = self.request(
-            "posts",
-            {
-                "tags": tags,
-                "per_page": per_page,
-                "page": page,
-                "tags_exclude": tags_exclude,
-                "exclude": exclude,
-                "categories": categories,
-                "group": groups,
-                "sticky": sticky,
-                "before": before,
-                "after": after,
-                "author": author,
-            },
-        )
+        try:
+            response = self.request(
+                "posts",
+                {
+                    "tags": tags,
+                    "per_page": per_page,
+                    "page": page,
+                    "tags_exclude": tags_exclude,
+                    "exclude": exclude,
+                    "categories": categories,
+                    "group": groups,
+                    "sticky": sticky,
+                    "before": before,
+                    "after": after,
+                    "author": author,
+                },
+            )
+        except HTTPError as error:
+            if error.response.status_code == 400:
```

Comment:
what about other codes? will they just silently break?

Reply:
Other 40X errors not likely to occur (e.g. if 403 happens then it should fail and not be handled because there must have been some change on server side that we need to update like tokens or secrets or firewall, or 402 similar error codes forbidden access are unusual and indicate something changed) this is a 400 bad request which is returned by the API endpoint and we need to handle as a result.

500 errors should break and they will show up in Sentry for us to fix and people should report it.

Reply:
I would have wanted to even skip the `status_code` but then it fails because there is no json. So I wanted to be as specific as possible to avoid catching unwanted errors

Reply:
okay, if you are sure that in all other error cases than 400 this will still work as expected, meaning it will report to Sentry and there won't be any weird behavior on the client side, then good to go

Reply:
Yes, in this specific case, we don't want it to break, because it's just somebody randomly entering a number that exceeds the limit. In theory WordPress API should return a 404, but for some reason they API specification doesn't determine that is how it should be.